node/actors/control_api.go

```diff
@@ -1,26 +1,197 @@
 package actors
 
 import (
+	"errors"
+	"fmt"
+	"log/slog"
+
 	"ergo.services/ergo/act"
 	"ergo.services/ergo/gen"
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
 )
 
-func createControlAPIServer() gen.ProcessBehavior {
-	return &controlAPIServer{}
+const APIPrefix = "$NEX"
+
+func createControlAPI() gen.ProcessBehavior {
+	return &controlAPI{
+		subsz: make([]*nats.Subscription, 0),
+	}
 }
 
-type controlAPIServer struct {
+type controlAPI struct {
 	act.Actor
+
+	publicKey  string
+	publicXKey string
+	subsz      []*nats.Subscription
 }
 
-func (api *controlAPIServer) Init(args ...any) error {
-	api.Log().Info("Control API Server started")
+func (api *controlAPI) Init(args ...any) error {
+	if len(args) != 2 {
+		err := errors.New("NATS connection and public key are required")
+		api.Log().Error("Failed to start control API", slog.String("error", err.Error()))
+		return err
+	}
+
+	if _, ok := args[0].(*nats.Conn); !ok {
+		err := errors.New("args[0] must be a valid NATS connection")
+		api.Log().Error("Failed to start control API", slog.String("error", err.Error()))
+		return err
+	}
+
+	if publicKey, ok := args[1].(string); ok {
+		api.publicKey = publicKey
+	}
+
+	err := api.initPublicXKey()
+	if err != nil {
+		return err
+	}
+
+	err = api.subscribe(args[0].(*nats.Conn))
+	if err != nil {
+		_ = api.shutdown()
+		return err
+	}
+
+	api.Log().Info("Control API started",
+		slog.String("public_key", api.publicKey),
+		slog.String("public_xkey", api.publicXKey),
+	)
 
 	return nil
 }
 
 // HandleInspect invoked on the request made with gen.Process.Inspect(...)
-func (api *controlAPIServer) HandleInspect(from gen.PID, item ...string) map[string]string {
+func (api *controlAPI) HandleInspect(from gen.PID, item ...string) map[string]string {
 	api.Log().Info("control API server got inspect request from %s", from)
 	return nil
 }
+
+func (api *controlAPI) initPublicXKey() error {
+	kp, err := nkeys.CreateCurveKeys()
+	if err != nil {
+		api.Log().Error("Failed to create x509 curve key", slog.Any("error", err))
+		return nil
+	}
+
+	publicXKey, err := kp.PublicKey()
+	if err != nil {
+		api.Log().Error("Failed to get public key from x509 curve key", slog.Any("error", err))
+		return nil
+	}
+
+	api.publicXKey = publicXKey
+	return nil
+}
+
+func (api *controlAPI) shutdown() error {
+	var err error
+
+	for _, sub := range api.subsz {
```

Comment:
This was a bad pattern before and we shouldn't copy it over. Call `Drain` once on the connection.

Reply:
If we don't drain these subscriptions and the actor restarts, we will end up with duplicate subscriptions. The underlying NATS connection is used for other things in the node so we shouldn't call `Drain` on it from the control API-- we do need to call `Drain` on it though during node shutdown and I will add that to this PR.

Also, it is unclear when the `controlAPI` actor would actually restart, since there isn't anything actually running in the background. We should clarify this more broadly when it comes to some of the other actors as well that seemingly will never exit unexpectedly.

Reply:
the actor restart policy is declared explicitly by their supervisor. Manually invoking `shutdown` like this implies that you're shutting it all down and that stuff isn't going to come back... which means we can do a connection-wide drain.

If we were only draining the stuff for controlAPI because we thought it was coming back up, then the subsz makes sense. But, as written this kind of shutdown is permanent.

tl;dr:
* if the shutdown is going to be a restart, use subsz (doesn't look like this is the case)
* if the shutdown is permanent, use connection-wide single-call Drain

Reply:
This makes sense, @autodidaddict.

In the meantime, can you please address the note related to how some of the defined actors will actually never exit unexpectedly? (e.g., the internal NATS server, control API, and host services actors should never actually "crash")

We are still using the `Actor` pattern for these structs. Was the rational primarily to support ergo observability for these?

Reply:
If one of those actors has a panic, ergo will halt that actor. Observability is like a free bonus. The main reason for ergo is so that we don't have to write a single mutex in the code anywhere.

Reply:
After thinking about this during lunch, the `subsz` pattern is probably safer, even if we don't expect to be bouncing those actors. Carry on :) 
![image](https://github.com/user-attachments/assets/eec79ee4-05b9-47ae-97f8-a402292bd5e2)